### ISSUE TITLE
Error when 500 template contains a django-allauth template tag

### ISSUE THREAD

**Issue by Bartleby2718** [pennersr/django-allauth#2387]
**Error message**: 
- `AttributeError: 'NoneType' object has no attribute 'POST'`

**How to reproduce**:
1) Create a 500 template (`500.html` in your template directory) that includes a template tag from django-allauth. For me, it was `Google Sign In` button:
```html
<!--500.html-->
{% load socialaccount %}
<a class="nav-link" href="{% provider_login_url 'google' %}">Log In</a>
```

2. Add an endpoint that is handled by Django's default 500 handler (`handler500` in `django.conf.urls`, which by default points to `django.views.defaults.server_error`)
```python
# urls.py
from django.conf.urls import handler500
urlpatterns = [
    # ...
    path('500/', handler500, name='500'),
    # ...
]
```
- The handler (`server_error`) renders the template: `return HttpResponseServerError(template.render())`
- `render` in `allauth/socialaccount/templatetags/socialaccount.py` is called
- `get_request_param` in `allauth/utils.py` is called in the line `next = get_request_param(request, 'next')`
- `return request.POST.get(param) or request.GET.get(param, default)` in `get_request_param` causes an error because `request` is `None` in this case

**Solution**:
- Add a guard statement to the function `get_request_param` in `allauth/utils.py`
- before a patch: Use a custom 500 handler instead of Django's default 500 handler

**Comment by pennersr** [comment]
Closed via 696a840d

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2387
